pkg/apis/monitoring/v1alpha1/scrapeconfig_types.go

```diff
@@ -359,3 +367,50 @@ type EC2SDConfig struct {
 	// +optional
 	Port *int `json:"port"`
 }
+
+type ProxyConfig struct {
+	// HTTP proxy server to use to connect to the targets.
+	//
+	// It requires Prometheus >= v2.43.0.
+	// +optional
+	ProxyURL *string `json:"proxyUrl,omitempty"`
```

Comment:
Does it need to be a URL format?
If so we can use kube builder pattern annotation to add a URL regex, WDYT?